src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -504,13 +504,33 @@ public function iShouldNotBeAbleToConfirmOrderBecauseDoNotBelongsToShippingCateg
     }
 
     /**
-     * @Then I should not be able to select :paymentMethodName payment method
+     * @Then I should not be able to select :paymentMethod payment method
      */
-    public function iShouldNotBeAbleToSelectPaymentMethod(string $paymentMethodName): void
+    public function iShouldNotBeAbleToSelectPaymentMethod(PaymentMethodInterface $paymentMethod): void
     {
-        $paymentMethods = $this->getPossiblePaymentMethods();
+        $this->iChoosePaymentMethod($paymentMethod);
 
-        Assert::false(array_search($paymentMethodName, array_column($paymentMethods, 'name'), true));
+        Assert::true(
+            $this->responseChecker->hasViolationWithMessage($this->ordersClient->getLastResponse(),
```

Comment:
```suggestion
            $this->responseChecker->hasViolationWithMessage(
                $this->ordersClient->getLastResponse(),
```

Well I might be wrong with this comment, but when checked some other `... hasViolationWithMessage...` steps they are like this 😃

Reply:
for me both options are fine, but it fit in a single line so I decided to left it in one line :)